internal/engine/image_builder.go

```diff
@@ -43,6 +83,25 @@ func (icb *imageBuilder) CanReuseRef(ctx context.Context, iTarget model.ImageTar
 func (icb *imageBuilder) Build(ctx context.Context, iTarget model.ImageTarget,
 	ps *build.PipelineState) (refs container.TaggedRefs, err error) {
 	userFacingRefName := container.FamiliarString(iTarget.Refs.ConfigurationRef)
+	startTime := time.Now()
+	ctx, err = tag.New(ctx, tag.Upsert(KeyImageRef, userFacingRefName))
+	if err != nil {
+		return container.TaggedRefs{}, err
+	}
+
+	defer func() {
```

Comment:
we're probably going to want helper functions for a lot of this stuff soon to make it easy to drop into various functions, but this is good for now